src/lib/support/BytesToHex.cpp

```diff
@@ -72,6 +73,34 @@ CHIP_ERROR MakeU8FromAsciiHex(const char * src, const size_t srcLen, uint8_t * v
     return CHIP_NO_ERROR;
 }
 
+size_t HexToBytes(const char * src_hex, const size_t src_size, uint8_t * dest_bytes, size_t dest_size_max, BitFlags<HexFlags> flags)
+{
+    if ((src_hex == nullptr) || (dest_bytes == nullptr))
+    {
+        return 0;
+    }
+    // Octet string where each octet is 2 ascii digits representing the hex value
+    // Each is represented by two ascii chars, so must be even number
+    if ((src_size & 0x1) != 0 || src_size > dest_size_max * 2)
+    {
+        return 0;
+    }
+
+    memset(dest_bytes, 0, dest_size_max);
+    size_t bytesFilled = 0;
+
+    for (size_t i = 0; i < src_size; i += 2)
+    {
+        if (MakeU8FromAsciiHex(src_hex + i, 2, &dest_bytes[i / 2], flags) != CHIP_NO_ERROR)
+        {
+            memset(dest_bytes, 0, dest_size_max);
```

Comment:
Not sure what we gain by clearing memory here. Please consider not doing so.

Reply:
I removed that - it is not really needed.